src/adapter/src/optimize/materialized_view.rs

```diff
@@ -270,6 +275,9 @@ impl Optimize<LocalMirPlan> for Optimizer {
             &self.typecheck_ctx,
             &mut df_meta,
         );
+        for id in self.force_non_monotonic.iter() {
+            transform_ctx.set_force_non_monotonic(*id);
+        }
```

Comment:
Instead of plumbing this through the `TransformCtx`, can we just modify the monotonicity of the sources here directly? E.g.:

```rust
for id in &self.force_non_monotonic {
    if let (desc, monotonic) = df_desc.source_imports.get_mut(id) {
        *monotonic = false;
    }
}
```